### ISSUE TITLE
cannot find package six

### ISSUE THREAD

**Issue by digital-nomad17** [ansible/ansible#24628]
Hi,

I just created a fresh python2 virtualenv and installed ansible. When I try to run trellis playbook, I get a 'cannot find compat.six module' error. I ran 'pip install six' and it's already installed. anyone else having this issue with the latest ansible?

##### ISSUE TYPE
missing module - 'no module named six'

##### COMPONENT NAME
six
##### ANSIBLE VERSION
ansible 2.4.0
python 2.7.3

##### OS / ENVIRONMENT
mac os 10.12.4

##### SUMMARY
I just created a fresh python2 virtualenv and installed ansible. When I try to run trellis playbook, I get a 'cannot find compat.six module' error. I ran 'pip install six' and it's already installed. anyone else having this issue with the latest ansible?

traceback:
```
ERROR! Unexpected Exception: No module named six
the full traceback was:

Traceback (most recent call last):
  File "/Users/andrewcarre/wordpress-python/wordpress_py2/bin/ansible-playbook", line 109, in <module>
    exit_code = cli.run()
  File "/Users/andrewcarre/wordpress-python/wordpress_py2/lib/python2.7/site-packages/ansible/cli/playbook.py", line 158, in run
    results = pbex.run()
  File "/Users/andrewcarre/wordpress-python/wordpress_py2/lib/python2.7/site-packages/ansible/executor/playbook_executor.py", line 89, in run
    self._tqm.load_callbacks()
  File "/Users/andrewcarre/wordpress-python/wordpress_py2/lib/python2.7/site-packages/ansible/executor/task_queue_manager.py", line 179, in load_callbacks
    for callback_plugin in callback_loader.all(class_only=True):
  File "/Users/andrewcarre/wordpress-python/wordpress_py2/lib/python2.7/site-packages/ansible/plugins/__init__.py", line 415, in all
    self._module_cache[path] = self._load_module_source(name, path)
  File "/Users/andrewcarre/wordpress-python/wordpress_py2/lib/python2.7/site-packages/ansible/plugins/__init__.py", line 338, in _load_module_source
    module = imp.load_source(name, path, module_file)
  File "/Users/andrewcarre/Projects/carre.ninja/trellis/lib/trellis/plugins/callback/vars.py", line 8, in <module>
    from ansible.compat.six import iteritems
ImportError: No module named six
```
```
(wordpress_py2) Andrews-MacBook-Pro:trellis andrewcarre$ ansible --version
ansible 2.4.0
  config file = /Users/andrewcarre/Projects/carre.ninja/trellis/ansible.cfg
  configured module search path = [u'/Users/andrewcarre/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/andrewcarre/wordpress-python/wordpress_py2/lib/python2.7/site-packages/ansible
  executable location = /Users/andrewcarre/wordpress-python/wordpress_py2/bin/ansible
  python version = 2.7.13 (default, Apr  4 2017, 08:47:57) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]
```

**Comment by sivel** [comment]
It seems that you have a callback plugin located at the following path, that is not compatible with ansible v2.4.0:

```
/Users/andrewcarre/Projects/carre.ninja/trellis/lib/trellis/plugins/callback/vars.py
```

Since I see you are using the following ansible.cfg, I am guessing you are have configured ansible to look at that callback plugins directory:

```
/Users/andrewcarre/Projects/carre.ninja/trellis/ansible.cfg
```

needs_info

**Comment by digital-nomad17** [comment]
here is my ansible.cfg file:
```
[defaults]
callback_plugins = ~/.ansible/plugins/callback_plugins/:/usr/share/ansible_plugins/callback_plugins:lib/trellis/plugins/callback
stdout_callback = output
filter_plugins = ~/.ansible/plugins/filter_plugins/:/usr/share/ansible_plugins/filter_plugins:lib/trellis/plugins/filter
force_color = True
force_handlers = True
inventory = hosts
nocows = 1
roles_path = vendor/roles
vars_plugins = ~/.ansible/plugins/vars_plugins/:/usr/share/ansible_plugins/vars_plugins:lib/trellis/plugins/vars
vault_password_file = .vault_pass

[ssh_connection]
ssh_args = -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s
pipelining = True
```
I am using the latest version of [trellis](https://github.com/roots/trellis) and it says it's compatible with all versions above 2.2.

**Comment by sivel** [comment]
You are running ansible v2.4.0 (devel) which has not yet been released from within the directory configured for trellis.  The file the trellis callback is trying to import no longer exists.

You will need to install ansible v2.2 or v2.3 for use with trellis until it is updated accordingly.

If you have further questions please stop by IRC or the mailing list:
* IRC: #ansible on irc.freenode.net
* mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Comment by s3w47m88** [comment]
I am experiencing this issue now as well.

    ansible 2.4.0.0
      config file = /home/{MYUSER}/Sites/{MYSITE}.com/trellis/ansible.cfg
      configured module search path = [u'/home/{MYUSER}/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
      ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
      executable location = /usr/local/bin/ansible
      python version = 2.7.14 (default, Sep 23 2017, 22:06:14) [GCC 7.2.0]
`

Since 2.4.x is released, and Trellis now requires 2.4.x is this a bug?

**Comment by omani** [comment]
I guess this is actually a bug. I have the same issue. ansible 2.4.3.0 here.
